clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchResponse.java

```diff
@@ -110,10 +149,26 @@ public OffsetFetchResponse(Struct struct) {
                 long offset = partitionResponse.getLong(COMMIT_OFFSET_KEY_NAME);
                 String metadata = partitionResponse.getString(METADATA_KEY_NAME);
                 short errorCode = partitionResponse.getShort(ERROR_CODE_KEY_NAME);
+                if (errorCode != Errors.NONE.code() && !PARTITION_ERRORS.contains(errorCode))
+                    topLevelErrorCode = errorCode;
                 PartitionData partitionData = new PartitionData(offset, metadata, errorCode);
                 responseData.put(new TopicPartition(topic, partition), partitionData);
             }
         }
+
+        this.errorCode = struct.hasField(ERROR_CODE_KEY_NAME) ? struct.getShort(ERROR_CODE_KEY_NAME) : topLevelErrorCode;
+    }
+
+    public short errorCode() {
```

Comment:
Definitely. This is one of my favorite gripes. Using more specific types whenever possible allows the compiler to do more work for us.